src/pytkdocs/loader.py

```diff
@@ -405,15 +405,18 @@ def get_class_documentation(self, node: ObjectNode, select_members=None) -> Clas
             self.select_inherited_members and "__fields__" in all_members
         ):
             root_object.properties = ["dataclass"]
-            for field_name, annotation in all_members["__annotations__"].items():
-                if self.select(field_name, select_members) and (  # type: ignore
-                    self.select_inherited_members
-                    # Same comment as for Pydantic models
-                    or field_name
-                    not in chain(*(getattr(cls, "__dataclass_fields__", {}).keys() for cls in class_.__mro__[1:-1]))
-                ):
-                    child_node = ObjectNode(obj=annotation, name=field_name, parent=node)
-                    root_object.add_child(self.get_annotated_dataclass_field(child_node))
+            # In dataclasses: "A field is defined as class variable that has a type annotation"
+            # so we can get all fields simply by iterating through the __annotations__
+            if "__annotations__" in all_members:
+                for field_name, annotation in all_members["__annotations__"].items():
```

Comment:
Actually, why don't we iterate on `__dataclass_fields__` since we already asserted its existence? Besides, the dataclass field also contains a `default` value, which we could use for https://github.com/pawamoy/pytkdocs/issues/50

Reply:
That's a better idea! Updated.

As far as I can tell, actually using the default value would be a bit more work since it would have to be added to the `Attribute` object and then presumably used in some serialisation somewhere but I can't immediately see where. So I guess that can happen separately :)

Reply:
You are absolutely right :slightly_smiling_face: I will merge this PR as it is, thanks!